### PR TITLE
Fix g++ 10 compile errors.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
For issue #6716.

Two commits.
* Trivial changes to fix errors.
* Redo `traffic_via` to use `string_view`, which fixes an error which would otherwise be more painful to deal with.

### CODE REVIEW

File: src/traffic_via/traffic_via.cc [hunk 1]
```diff
@@ -26,9 +26,12 @@
 #include "tscore/I_Version.h"
 #include "tscore/Tokenizer.h"
 #include "tscore/TextBuffer.h"
+//#include "tscpp/util/TextView.h"
```
Comment (bryancall) [inline src/traffic_via/traffic_via.cc:29]: I would remove this commented line, besides that it looks good.

File: iocore/cache/P_CacheTest.h [hunk 2]
```diff
@@ -44,7 +44,7 @@ struct PinnedDocTable : public Continuation {
   int remove(CacheKey *key);
   int cleanup(int event, Event *e);
 
-  PinnedDocTable() : Continuation(new_ProxyMutex()) { memset(bucket, 0, sizeof(Queue<PinnedDocEntry>) * PINNED_DOC_TABLE_SIZE); }
+  PinnedDocTable() : Continuation(new_ProxyMutex()) { ink_zero(bucket); }
```
Comment (maskit) [inline iocore/cache/P_CacheTest.h:47]: Do we need this initialization? `Queue`'s constructor should be called as long as this constructor is called.

File: proxy/http/HttpTunnel.cc [hunk 3]
```diff
@@ -479,8 +479,8 @@ HttpTunnel::reset()
 
   num_producers = 0;
   num_consumers = 0;
-  memset(consumers, 0, sizeof(consumers));
-  memset(producers, 0, sizeof(producers));
+  ink_zero(consumers);
```
Comment (maskit) [inline proxy/http/HttpTunnel.cc:482]: I don't think this is a right way to solve the problem. I'd use placement new for each item.

**Comment by SolidWallOfCode** [comment]
@maskit I understand your points, but I think those should be separate PRs. For this one, I tried to minimize changes. Unfortunately I had to do some non-trivial reworking in `traffic_via`, but I'd rather leave these as is and fix them independently.
@bryancall I'll get that fixed.

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**